src/components/tasks/card/TaskTagEdit.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from "react";
+
+import classNames from '@/components/tasks/card/card.module.scss';
+import { useTasksContext } from '@/context/tasks.context';
+import { TAG_TYPE, ITEM_TYPE } from '@/components/constants/items';
+import levelType from "@/interfaces/level.type";
+import tagType from "@/interfaces/tag.type";
+import { toast,ToastTypes } from "@/helperFunctions/toast";
+
+type TaskTagPropsType = {
+    updateTaskTagLevel: (taskItemToUpdate: any, method: 'delete' | 'post') => void
+}
+
+type SelectComponentPropsType = {
+    options: levelType[] | tagType[]
+    name: 'tags' | 'levels'
+    defaultOption: '--new tag--' | '--new level--'
+    setNewValueOnChange: React.Dispatch<React.SetStateAction<string | undefined>>
+    id: string
+    label: string
+    value: string | undefined
+}
+
+
+const SelectComponent = ({ label,
+    id,
+    setNewValueOnChange,
+    defaultOption,
+    options,
+    name,
+    value }: SelectComponentPropsType) => {
+
+return (
+    <div>
+    <label htmlFor={id}><span className={classNames.screenReaderOnly}>{label}</span></label>
+    <div className={classNames.selectWrapper}>
+    <select id={id}
+        name={name}
+        value={value}
+        onChange={(e) =>{
+            setNewValueOnChange(e.target.value)
+        }}
+        className={classNames.selectDropdown}>
+    <option disabled selected>{defaultOption}</option>
+    {
+        options?.map(option => (
+            <option key={option.name} value={option.name}>
+                {name === "levels" 
+                    ? `Level - ${option.name}`
+                    : option.name
+                }
+            </option>
+            ))
+    }
+    </select>
```

Comment:
can you do some formatting here, so that we know which is outermost and inner elements

Reply:
okay